Draft for guide doc (#811)
Samsung/ONE-vscode: docs/HowToInstall.md

# How to install?

## Prerequisit

ONE-vscode is an [extension](https://code.visualstudio.com/docs/editor/extension-marketplace) of [vscode](https://code.visualstudio.com/). To use ONE-vscode, you should install vscode first.

### vscode and its extensions

Let's install vscode according to [doc](https://code.visualstudio.com/docs/setup/setup-overview). (You should also install npm for vscode.)

### ssh extension in vscode

Now [ONE](https://github.com/Samsung/ONE) officially releases on linux/ubuntu only. If your develop environment is on Windows, you should use ssh extension in vscode. Let's follow [tutorial](https://code.visualstudio.com/docs/remote/ssh-tutorial).

### workspace

> Prepare workspace according to [doc](https://code.visualstudio.com/docs/editor/workspaces). ONE-vscode has been devloped based on workspace.

Reviewer: ```suggestion
Prepare workspace according to the [document](https://code.visualstudio.com/docs/editor/workspaces). ONE-vscode has been developed based on workspace.
```

I'm not sure it's good or bad using abbreviation in official document. (I don't have any preference but just for curiosity)

> # How to install?

Reviewer 2: Technically, 'how to install' is noun clause.
Maybe better to use something like the following:

- How to install

or 

- How to install vscode extension for ONE

## Prerequisit

ONE-vscode is an [extension](https://code.visualstudio.com/docs/editor/extension-marketplace) of [vscode](https://code.visualstudio.com/). To use ONE-vscode, you should install vscode first.

### vscode and its extensions

> Let's install vscode according to [doc](https://code.visualstudio.com/docs/setup/setup-overview). (You should also install npm for vscode.)

Reviewer 2: ```suggestion
Let's install vscode according to [doc](https://code.visualstudio.com/docs/setup/setup-overview). (You should also install npm for vscode.)
The workspace for one-vscode will have model files, e.g., *.tflite, *.onnx, etc., and ONE *.cfg files. Also intermediate files during compilation (e.g., *.circle) or compiled output files could also exist in your workspace.
```

### ssh extension in vscode

Now [ONE](https://github.com/Samsung/ONE) officially releases on linux/ubuntu only. If your development environment is on Windows, you should use ssh extension in vscode. Let's follow [tutorial](https://code.visualstudio.com/docs/remote/ssh-tutorial).

### workspace

> Prepare workspace according to the [document](https://code.visualstudio.com/docs/editor/workspaces). ONE-vscode has been developed based on workspace.

Reviewer 2: ```suggestion
Prepare workspace according to the [document](https://code.visualstudio.com/docs/editor/workspaces). ONE-vscode has been developed based on workspace.
The workspace for ONE-vscode will have model files, e.g., *.tflite, *.onnx, etc., and ONE *.cfg files. Also intermediate files during compilation (e.g., *.circle) or compiled output files could also exist in your workspace.
```

# How to install?

## Prerequisit

ONE-vscode is an [extension](https://code.visualstudio.com/docs/editor/extension-marketplace) of [vscode](https://code.visualstudio.com/). To use ONE-vscode, you should install vscode first.

### vscode and its extensions

> Let's install vscode according to [doc](https://code.visualstudio.com/docs/setup/setup-overview). (You should also install npm for vscode.)

Reviewer 3: What about this url `https://code.visualstudio.com/download` ?

### ssh extension in vscode

Now [ONE](https://github.com/Samsung/ONE) officially releases on linux/ubuntu only. If your development environment is on Windows, you should use ssh extension in vscode. Let's follow [tutorial](https://code.visualstudio.com/docs/remote/ssh-tutorial).

### workspace

Prepare workspace according to the [document](https://code.visualstudio.com/docs/editor/workspaces). ONE-vscode has been developed based on workspace.
Typically, a workspace for ONE-vscode has the following files:
- your model files
  - Tensoflow model file(*.pb, saved model or Keras model)
  - Tensorflow Lite model (*.tflite) 
  - Onnx model (*.onnx)
- ONE *.cfg files
- intermediate files during compilation (e.g., *.circle) or compiled output files after compilation

## How to install?

> If you are a user not a developer, installing ONE-vscode is enough. Let's install ONE-vscode with vsix file.

Reviewer 3: (Q) What is a difference between `a user` and `a developer` in this sentence?

Author: My point was
- a user: just uses ONE-vscode
- a developer: developed ONE-vscode

# How to install?

> ## Prerequisit

Reviewer: ```suggestion
## Prerequisite
```
maybe..?

ONE-vscode is an [extension](https://code.visualstudio.com/docs/editor/extension-marketplace) of [vscode](https://code.visualstudio.com/). To use ONE-vscode, you should install vscode first.

### vscode and its extensions

Let's install vscode.
- You can download vscode [HERE](https://code.visualstudio.com/download)
- Official document: [setup](https://code.visualstudio.com/docs/setup/setup-overview)
- (You should also install npm for vscode)

### ssh extension in vscode

Now [ONE](https://github.com/Samsung/ONE) officially releases on linux/ubuntu only. If your development environment is on Windows, you should use ssh extension in vscode. Let's follow [tutorial](https://code.visualstudio.com/docs/remote/ssh-tutorial).

> 

Reviewer 3: ```suggestion

### environment setting on linux/ubuntu

Linux machine must have `aptitude` and `docker` packages.

```

### workspace

Prepare workspace according to the [document](https://code.visualstudio.com/docs/editor/workspaces). ONE-vscode has been developed based on workspace.
Typically, a workspace for ONE-vscode has the following files:
- your model files

>   - Tensoflow model file(*.pb, saved model or Keras model)

Reviewer: ```suggestion
  - Tensorflow model file(*.pb, saved model or Keras model)
```

  - Tensorflow Lite model (*.tflite)
  - Onnx model (*.onnx)
- ONE *.cfg files
- intermediate files during compilation (e.g., *.circle) or compiled output files after compilation

## How to install?

If you are a user(who just wants to use ONE-vscode not to develop ONE-vscode), installing ONE-vscode is enough. Let's install ONE-vscode with vsix file.

### Installing with GUI

Extensions(Ctrl+Shift+x) -> Views and More Actions -> Install from VSIX

<img src="https://user-images.githubusercontent.com/10216715/174781792-fa3fbe47-708b-4977-974c-578b65f689ee.gif" width=500 />

### Installing with CLI

> ```
> $ ls
> one-vscode-0.1.0.vsix
> 
> $ code --install-extension ./one-vscode-0.1.0.vsix

Reviewer: ```suggestion
```console
$ ls
one-vscode-0.1.0.vsix

$ code --install-extension ./one-vscode-0.1.0.vsix
```

# How to install?

## Prerequisite

ONE-vscode is an [extension](https://code.visualstudio.com/docs/editor/extension-marketplace) of [vscode](https://code.visualstudio.com/). To use ONE-vscode, you should install vscode first.

### vscode and its extensions

Let's install vscode.
- You can download vscode [HERE](https://code.visualstudio.com/download)
- Official document: [setup](https://code.visualstudio.com/docs/setup/setup-overview)

> - (You should also install npm for vscode)

Reviewer 2: I am not sure if end-user need to install npm. Do they need npm?

Author: You're right. I'll remove it soon. 👍

### ssh extension in vscode

Now [ONE](https://github.com/Samsung/ONE) officially releases on linux/ubuntu only. If your development environment is on Windows, you should use ssh extension in vscode. Let's follow [tutorial](https://code.visualstudio.com/docs/remote/ssh-tutorial).

### environment setting on linux/ubuntu

> Linux machine must have `aptitude` and `docker` packages.

Reviewer 2: ```suggestion
Linux machine must have `aptitude`.
```

and `aptitude` is installed by toolchain view. 
@jyoungyun, could you check if we need this statement?

Author: After @jyoungyun 's check, I'll merge.
